src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,501 @@
+    """This method is used to traverse through the root VSS node to build
+       -> ApiGear equivalent string buffer and to serialize it accordingly into a file
+    """
+    module = ApiGearModule()
+    module_name = root.name
+    module_filename = MODULE_FILENAME.format(module_name)
+    module_path = os.path.join(directory, module_filename)
+
+    log.info(f"directory: {directory}")
+    log.info(f"Module name: {module_name}")
+    log.info(f"Module filename: {module_filename}")
+    log.info(f"Module Path: {module_path}")
+
+    if (not layers):
+        layers = {SolutionLayers.CPP: "../cppservice"}
+        log.warning(f"No layers provided! Defaulting to CPP in {layers[SolutionLayers.CPP]}")
+
+    log.debug(layers)
+
+    export_node(root, module)
+    export_data_type_node(data_type_tree, module)
+    generate_solution(directory, module_filename, module_name, layers)
+    with open(module_path, "w") as file:
+        generate_module(file, module, module_name)
+        log.info(f"Generated module file at {module_path}")
+
+    log.info("Apigear files generated at location : " + file.name)
+
+
+@click.command()
+@clo.vspec_opt
+@clo.output_required_opt
+@clo.include_dirs_opt
+@clo.extended_attributes_opt
+@clo.strict_opt
+@clo.aborts_opt
+@clo.uuid_opt
+@clo.expand_opt
+@clo.overlays_opt
+@clo.quantities_opt
+@clo.units_opt
+@clo.types_opt
+@clo.types_output_opt
+@click.option(
+    "--apigear-template-unreal-path",
+    required=False,
+    is_flag=False,
+    help="Add Unreal layer to solution file at the specified path",
+)
+@click.option(
+    "--apigear-template-cpp-path",
+    required=False,
+    is_flag=False,
+    help="Add Cpp14 layer to solution file at the specified path",
+)
+@click.option(
+    "--apigear-template-qt5-path",
+    required=False,
+    is_flag=False,
+    help="Add Qt5 layer to solution file at the specified path",
+)
+@click.option(
+    "--apigear-template-qt6-path",
+    required=False,
+    is_flag=False,
+    help="Add Qt6 layer to solution file at the specified path",
+)
+def cli(
+    vspec: Path,
+    output: Path,
+    include_dirs: tuple[Path],
+    extended_attributes: tuple[str],
+    strict: bool,
+    aborts: tuple[str],
+    uuid: bool,
+    expand: bool,
+    overlays: tuple[Path],
+    quantities: tuple[Path],
+    units: tuple[Path],
+    types: tuple[Path],
+    types_output: Path,
+    apigear_template_unreal_path: str,
+    apigear_template_cpp_path: str,
+    apigear_template_qt5_path: str,
+    apigear_template_qt6_path: str,
+):
+    """
+    Export as ApiGear.
+    """
+    tree, data_type_tree = get_trees(
+        include_dirs,
+        aborts,
+        strict,
+        extended_attributes,
+        uuid,
+        quantities,
+        vspec,
+        units,
+        types,
+        types_output,
+        overlays,
+        expand,
+    )
+    log.info("Generating ApiGear output...")
+    if os.path.exists(output):
+        log.error("Directory already exists. Aborting..")
```

Comment:
Output is not a directory but a file

Reply:
Output for ApiGear is actually a directory where the module and solution files will be placed.

Reply:
Well then you need a different argument. I think we have an --output-dir argument already in some exporter (protobuf?). So you could use that one or put it in cli_options and reuse it in both places

Reply:
Good call, I just wanted to keep it consistent across all of the other exporters. Protobuf actually uses the `--output` and `--types-out-dir` options differently, so I created a new required argument `--output-dir` for this exporter.